src/java/picard/sam/CompareSAMs.java

```diff
@@ -501,6 +501,11 @@ private void reportDifference(Object o1, Object o2, final String label) {
         reportDifference(o1.toString(), o2.toString(), label);
     }
 
+    private String getKeyForRecord(final SAMRecord record) {
+        boolean t = record.getReadPairedFlag() && record.getSecondOfPairFlag();
```

Comment:
Should be final.  And it would be nice to get a more informative variable name.  isSecondOfPair?